docs/tutorial.md

```diff
@@ -329,1206 +328,1106 @@ Connection: keep-alive
 
-[\[Get the Code for This Lesson\]](https://github.com/reimagined/resolve/tree/master/examples/shopping-list-tutorial/lesson-6)
-
-Currently, your Shopping List application's UI only allows users to view and edit one shopping list (**shopping-list-1**). This lesson describes how to create multiple shopping lists and navigate between these lists from the client UI.
+[\[Get the Code for This Lesson\]](https://github.com/reimagined/resolve/tree/master/examples/shopping-list-tutorial/lesson-4)
 
-### Implement a Shopping Lists Read Model
+This lesson describes how you can use a View Model to obtain shopping list items and display them as a list within the client browser.
 
-First, modify the application's backend so it can provide information about all available shopping lists. You can implement a View Model to achieve this goal, but this approach is inefficient.
+A View Model is a reactive Read Model that is built on the fly for one or several aggregate IDs. A client can maintain a WebSocket connection with a resolve server to receive View Model data updates in real time.
 
-Consider a situation when your application runs in a production environment for a long time and end-users have created a large number of shopping lists. In such situation, a View Model's projection would apply every event from the store to the response object. This would result in a considerable performance overhead _on every request_.
+The downside is that View Models do not have persistent state and should be rebuilt on every query, so they are better suited for small data samples.
 
-Use a **[Read Model](read-side.md#read-models)** to overcome this limitation. The code sample below demonstrates the ShoppingLists Read Model's projection.
+### Create a Shopping List View Model
 
-**common/read-models/shopping_lists.projection.js:**
+Add a **shopping_list.projection.js** file in the **common/view-models** directory. Add the following code to this file:
 
-<!-- prettier-ignore-start -->
+**common/view-models/shopping_list.projection.js**
 
-[embedmd]:# (../examples/shopping-list-tutorial/lesson-6/common/read-models/shopping_lists.projection.js /^/   /\n$/)
 ```js
-import { SHOPPING_LIST_CREATED } from '../eventTypes'
+import { SHOPPING_LIST_CREATED, SHOPPING_ITEM_CREATED } from '../eventTypes'
 
+// A View Model's projection is defined in a format that is isomorphic with a Redux reducer format.
 export default {
-  Init: async store => {
-    await store.defineTable('ShoppingLists', {
-      indexes: {
-        id: 'string'
+  // The 'Init' function initializes the View Model's response object.
+  Init: () => ({
+    id: 'id',
+    name: 'unnamed',
+    list: [],
+  }),
+  // Below is a projection function. It runs on every event of the specified type, whose aggregate Id matches one of the Ids specified in the query.
+  // A View Model projection takes the response object and returns its updated version based on the event data.
```

Comment:
The 'response object' term seems a bit misleading to me. It's a view model's state. It is passed to the response only if it is the latest event to process, otherwise response is not ready yet. Maybe we should think on terminology here.

Reply:
I think, you are right. The key point is that its is a View Model's state, so we should call it so. I rewrote the comments and also added a part that clarifies the  relation between the state and the response:

> // When all events are applied, the built state is passed to the client in the response body.